rust/ql/test/library-tests/dataflow/local/main.rs

```diff
@@ -233,6 +233,14 @@ fn option_unwrap_or() {
     sink(s2.unwrap_or(source(47))); // $ hasValueFlow=47
 }
 
+fn option_unwrap_or_else() {
+    let s1 = Some(source(47));
+    sink(s1.unwrap_or_else(|| 0)); // $ hasValueFlow=47
+
+    let s2 = None;
+    sink(s2.unwrap_or_else(|| source(48))); // $ MISSING: hasValueFlow=48
```

Comment:
We have the model:
```
      - ["lang:core", "<crate::option::Option>::unwrap_or_else", "Argument[0].ReturnValue", "ReturnValue", "value", "manual"]
```
@hvitved are we lacking data flow support here, or have I got the syntax wrong?

Reply:
I'll take a look.

Reply:
@geoffw0 : I was not able to push directly to your branch, so I have opened https://github.com/geoffw0/ql/pull/6 to fix this.

Reply:
Thanks, looks great, I've merged it into this PR.